c/shared/source/ufowrite/ufowrite.c

```diff
@@ -649,6 +550,159 @@ static void writeBlueValues(ufwCtx h, abfPrivateDict *privateDict){
+        writeLine(h, buffer);
+        writeLine(h, "\t<key>com.adobe.type.postscriptFDArray</key>");
+        writeLine(h, "\t<array>");
+        for (j = 0; j < top->FDArray.cnt; j++) {
+            writeLine(h, "\t<dict>");
+            abfFontDict *fd = &h->top->FDArray.array[j];
+            writeLine(h, "\t<key>FontName</key>");
+            sprintf(buffer, "\t<string>%s</string>", fd->FontName.ptr);
+            writeLine(h, buffer);
+            writeLine(h, "\t<key>PaintType</key>");
+            sprintf(buffer, "\t<integer>%ld</integer>", fd->PaintType);
+            writeLine(h, buffer);
+            if (fd->FontMatrix.cnt == ABF_EMPTY_ARRAY) {
+                fd->FontMatrix.cnt = 6;
+                fd->FontMatrix.array[0] = 0.001;
+                fd->FontMatrix.array[1] = 0.0;
+                fd->FontMatrix.array[2] = 0.0;
+                fd->FontMatrix.array[3] = 0.001;
+                fd->FontMatrix.array[4] = 0.0;
+                fd->FontMatrix.array[5] = 0.0;
+            }
+            writeLine(h, "\t<key>FontMatrix</key>");
+            writeLine(h, "\t<array>");
+            for (i = 0; i < fd->FontMatrix.cnt; i++) {
+                float stem = fd->FontMatrix.array[i];
+                if (stem == ((int)stem))
+                    sprintf(buffer, "\t\t<integer>%d</integer>", (int)stem);
+                else
+                    sprintf(buffer, "\t\t<real>%.3f</real>", stem);
+                writeLine(h, buffer);
+            }
+            writeLine(h, "\t</array>");
+
+            privateDict = &(fd->Private);
+            writeLine(h, "\t<key>PrivateDict</key>");
+            writeLine(h, "\t<dict>");
+            writeBlueValues(h, privateDict);
+            writeLine(h, "\t</dict>");
+            writeLine(h, "\t</dict>");
+        }
+        writeLine(h, "\t</array>");
+    }
+    return 0;
+}
+
+static void writeGlyphOrder(ufwCtx h) {
+    char buffer[FILENAME_MAX];
+    int i;
+
+    /* Set error handler */
+    DURING_EX(h->err.env)
+
+    h->state = 1; /* Indicates writing to dst stream */
+
+    /* Open lib.plist file as dst stream */
+
+    sprintf(buffer, "%s", "lib.plist");
+    h->cb.stm.clientFileName = buffer;
+    h->stm.dst = h->cb.stm.open(&h->cb.stm, UFW_DST_STREAM_ID, 0);
+    if (h->stm.dst == NULL)
+        fatal(h, ufwErrDstStream);
+
+    writeLine(h, XML_HEADER);
+    writeLine(h, PLIST_DTD_HEADER);
+    writeLine(h, "<plist version=\"1.0\">");
+    writeLine(h, "<dict>");
+    writeLine(h, "\t<key>public.glyphOrder</key>");
+    writeLine(h, "\t<array>");
+    for (i = 0; i < h->glyphs.cnt; i++) {
+        Glyph *glyphRec;
+        glyphRec = &h->glyphs.array[i];
+        sprintf(buffer, "\t\t<string>%s</string>", glyphRec->glyphName);
+        writeLine(h, buffer);
+    }
+
+    writeLine(h, "\t</array>");
+
+    if (h->top->sup.flags & ABF_CID_FONT) {
+        if (h->top->cid.CIDFontName.ptr != NULL) {
+            writeLine(h, "\t<key>com.adobe.type.cid.CIDFontName</key>");
```

Comment:
Can we drop `cid` from all the keys so just `com.adobe.type.CIDFontName`, etc? Originally I was thinking `cid` would make it clearer, but the rest of the info already does that.